### PR TITLE
VMware: Add new module: vmware_host_service_manager

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME

lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py
test/integration/targets/vmware_host_service_manager/aliases
test/integration/targets/vmware_host_service_manager/tasks/main.yml

##### ANSIBLE VERSION

```
2.5devel
```

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    description:
+    - Name of the cluster.
+    - Service settings are applied to every ESXi host system/s in given cluster.
+    - If C(esxi_hostname) is not given, this parameter is required.
+  esxi_hostname:
+    description:
+    - ESXi hostname.
+    - Service settings are applied to this ESXi host system.
+    - If C(cluster_name) is not given, this parameter is required.
+  state:
```
Comment (pdellaert) [inline lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py:43]: Should there be a state to enable/disable the service on host boot?

File: lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                    self.results[host.name].update(changed=service_state)
+                except (vim.fault.InvalidState, vim.fault.NotFound, vim.fault.HostConfigFault) as e:
+                    self.results[host.name].update(changed=False,
+                                                   error=to_native(e.msg))
+
+        if any(host_service_state):
+            changed = True
+        self.module.exit_json(changed=changed, results=self.results)
+
+    def check_service_state(self, host, service_name):
```
Comment (pdellaert) [inline lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py:184]: Can be replaced with:
```python
def check_service_state(self, host, service_name):
    host_service_system = host.configManager.serviceSystem
    if host_service_system:
        services = host_service_system.serviceInfo.service
        for service in services:
            if service.key == service_name:
                return service.running

    msg = "Failed to find '%s' service on host system '%s'" % (service_name, host.name)
    cluster_name = self.params.get('cluster_name', None)
    if cluster_name:
        msg += " located on cluster '%s'" % cluster_name
    msg += ", please check if you have specified a valid ESXi service name."
    self.module.fail_json(msg=msg)
```
This way it stops looping through services as soon as it finds the correct one, and if it finds none, it fails

File: lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        self.results = {}
+
+    def service_ctrl(self):
+        changed = False
+        host_service_state = []
+        for host in self.hosts:
+            actual_service_state = self.check_service_state(host=host, service_name=self.service_name)
+            host_service_system = host.configManager.serviceSystem
+            if host_service_system:
+                service_state = False
```
Comment (pdellaert) [inline lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py]: Personally i'd like it better if this was `changed_state`, for instance, as that better indicates the purpose of this variable, looking at the code

**Review by pdellaert — COMMENTED** [review]
Nice addition, just some remarks/potential improvements

File: lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py:43]: Sounds good idea.

File: lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py (hunk 3 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py]: OK.

File: lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py (hunk 2 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py:184]: OK.

File: lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py (hunk 1 above)
Comment (tchernomax) [reply lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py:43]: It's now handle by the "service_policy" option.

**Review by tchernomax — APPROVED** [review]
LGTM

**Review by pdellaert — APPROVED** [review]
LGTM
shipit

**Comment by Akasurde** [comment]
@tchernomax @pdellaert Thanks for your valuable comments and reviews.

**Pull request merged**